src/main/scala/firrtl/Emitter.scala

```diff
@@ -454,7 +454,11 @@ class VerilogEmitter extends Emitter with PassBased {
           instdeclares += Seq(");")
           sx
         case sx: DefMemory =>
-          declare("reg", sx.name, VectorType(sx.dataType, sx.depth))
+          val fullSize = sx.depth*(sx.dataType match { case GroundType(i:IntWidth) => i.width })
+          if(fullSize > (1 << 29))
+            declare("reg /* sparse */", sx.name, VectorType(sx.dataType, sx.depth))
+          else
+            declare("reg", sx.name, VectorType(sx.dataType, sx.depth))
```

Comment:
since the if is only changing the first arg of declare I think it would be better if you rewrote this as something like:
```scala
val decl = if (fullSize > (1 << 29)) "reg /* sparse */" else "reg"
declare(decl, sx.name, VectorType(sx.dataType, sx.depth))
```